### PR TITLE
Document how to opt-out of LSP `gq` behaviour

### PR DESCRIPTION
**Pull request by reegnz**
Add documentation discussed in #22611

### CODE REVIEW

**Comment by reegnz** [comment]
For `formatexpr` I'm struggling a bit on how to link it to the LSP docs (or whether it's even appropriate to link to from there).

**Comment by noahfrederick** [comment]
If I came across this, I wouldn't know what to make of "you could try using" and "if that is not good enough for you". How about:

> A: To use the default internal formatting method and bypass the LSP client's 'formatexpr', use |gw| instead.
> Alternatively, you can completely disable LSP formatting with |gq| by unsetting 'formatexpr':
>
> ...
>
> This will cause |gq| to fall back to formatting internally (or using 'formatprg', if set).

What do you think?

File: runtime/doc/lsp.txt [hunk 1]
```diff
@@ -137,7 +137,18 @@ FAQ                                                     *lsp-faq*
      " Auto-format *.rs (rust) files prior to saving them
      " (async = false is the default for format)
      autocmd BufWritePre *.rs lua vim.lsp.buf.format({ async = false })
+<
+- Q: How can I disable LSP formatting when using the |gq| command?
+  A: First you could try using |gw| as an alternative to |gq| .
+  If that is not good enough for you, then you can use the following 
+  lua snippet to completely disable LSP formatting using |gq| :
```
Comment (justinmk) [inline runtime/doc/lsp.txt:144]: no need to link |gq| every time it is mentioned, use "gq" after the first time. Else it's visually noisy.

**Comment by justinmk** [comment]
lgtm after comments are addressed, can improve it later

File: runtime/doc/lsp.txt (hunk 1 above)
Comment (reegnz) [reply runtime/doc/lsp.txt]: Reduced noise.

**Comment by justinmk** [comment]
still in draft?

**Comment by reegnz** [comment]
Ah sorry my bad, didn't notice I left it in draft.

**Pull request merged**